### DISCUSSION TITLE
NativeAnimation in TouchBehavior

### DISCUSSION THREAD

**Discussion by christianrr** [CommunityToolkit/Maui#1886] [New Feature Discussions]
I noticed that Native Animation / `NativeAnimation` was not ported into the recent release of TouchBehavior. I think this was one of the greatest things about XCT's TouchEffect and I can't understand, why Maui in general does not implement a single method to show native touch feedback (e.g. Ripple in Android).

However, from what I've seen, the ported code was already there and principally working in the feature branch #1673.
@brminnick removed it in late March. Was this for a specific reason?

Are there any plans to bring the functionality back? I think this would be a super important feature.

**Comment by TheCodeTraveler** [comment]
> However, from what I've seen, the ported code was already there and principally working in the feature branch #1673.
> @brminnick removed it in late March. Was this for a specific reason?

The reason was simply that the `NativeAnimation` feature was not working and we couldn't figure it out. So we decided to remove it from the initial release of `TouchBehavior` to avoid further delaying its release.

I'm totally supportive to you submitting a PR adding `NativeAnimation` support to `TouchBehavior` 💯

**Reply by christianrr** [reply]
@brminnick Thanks for the info. What were the issues with the feature? In the conversation of #1673 @Axemasta  mentioned that he was able to fix the issue with the Ripple animation on Android!?

I've once added a pull request with a working version of `NativeAnimation` in #1326. From what I've seen this was rebased into #1673. However, I use "my" version of `TouchBehavior` currently on a production app and `NativeAnimation` works here.

**Comment by CliffAgius** [comment]
Closed as answered, if closed in error please let us know to re-open.
